### DISCUSSION TITLE
CLI - Allow Export by Album/Account

### DISCUSSION THREAD

**Discussion by creedda** [ente-io/ente#714] [Enhancements]
Feature request.

Please allow exporting individual albums rather than all or nothing. I would assume the same should be done for accounts (though I have not tested adding multiple accounts yet)

This is not something we would probably need regularly, but it is good for testing to make sure the export is going to work with a smaller number of files.

Probably would need to add an "ente-cli account album list" and an "ente-cli export -album *album name*" and "ente-cli export -account *account-name*" as options to allow granularity.

Thank you again!

**Comment by mxmehl** [comment]
I would love to have this feature as well. Ideally, this would be combined with an option to **exclude one or multiple albums** or those that are **shared with us**.

Background: I would like to export all photos of me and my family members. However, we share some albums with each other. As of now, I could somehow hack that there are multiple jobs for the different accounts, but each would export the sometimes huge shared albums that are present in all/most accounts.

**Reply by mxmehl** [reply]
I just noticed that you can add multiple accounts. Would this de-duplicate the albums already? I would test it but as both accounts use a lot of space it would take a looong time.

**Reply by vishnukvmd** [reply]
Sorry no, this does not de-duplicate albums currently.

It'll pull all files belonging to respective accounts.

**Reply by robertsattler** [reply]
For now, I set up an external tool for deduplication right after export, like this:

    ente export
    rdfind -minsize 50000 -makehardlinks true -makeresultsfile false ./ente

Removing 6 GB of duplicates took 2-3 minutes. I would expect rmlint to be faster, but the simplicity of rdfind made it an easy decision for now. Using -minsize 50000 to exclude JSON metadata files is a precaution. Executing `ente export` after `rdfind` again does not redownload any files.

Could you please tell whether there is risk to deduplicate a) the photos like this and b) the metadata files. Thank you.

**Reply by vishnukvmd** [reply]
Hey, not super familiar with `rdfind`, but from what I can see from their documentation, this looks good!

Since `rdfind` compares files based on the content, and you're skipping the metadata files, both a) and b) come across as risk free.

**Comment by ua741** [comment]
The latest version of the CLI now supports these filters.

**Comment by alexandre-walzberg** [comment]
Should be great to be able to export the "Favoris" album, as in the end this the photo you really want to backup and have in local copies (for exemple screensaver and wallpapers)...
Then I'd like to have a crontab that sync local folder with only the ente favoris album on a regular basis, without exporting all albums.

**Reply by ua741** [reply]
> Should be great to be able to export the "Favoris" album, as in the end this the photo you really want to backup and have in local copies (for exemple screensaver and wallpapers)... Then I'd like to have a crontab that sync local folder with only the ente favoris album on a regular basis, without exporting all albums.

Can you try exporting with `Favorites` and let us know if that works?

**Comment by Tyno1997** [comment]
aka.ms

**Comment by Tyno1997** [comment]
# Contributor Covenant Code of Conduct

## Our Pledge

We as members, contributors, and leaders pledge to make participation in our
community a harassment-free experience for everyone, regardless of age, body
size, visible or invisible disability, ethnicity, sex characteristics, gender
identity and expression, level of experience, education, socio-economic status,
nationality, personal appearance, race, caste, color, religion, or sexual
identity and orientation.

We pledge to act and interact in ways that contribute to an open, welcoming,
diverse, inclusive, and healthy community.

## Our Standards

Examples of behavior that contributes to a positive environment for our
community include:

* Demonstrating empathy and kindness toward other people
* Being respectful of differing opinions, viewpoints, and experiences
* Giving and gracefully accepting constructive feedback
* Accepting responsibility and apologizing to those affected by our mistakes,
  and learning from the experience
* Focusing on what is best not just for us as individuals, but for the overall
  community

Examples of unacceptable behavior include:

* The use of sexualized language or imagery, and sexual attention or advances of
  any kind
* Trolling, insulting or derogatory comments, and personal or political attacks
* Public or private harassment
* Publishing others' private information, such as a physical or email address,
  without their explicit permission
* Other conduct which could reasonably be considered inappropriate in a
  professional setting

## Our Responsibilities

Community leaders are responsible for clarifying and enforcing our standards of
acceptable behavior and will take appropriate and fair corrective action in
response to any behavior that they deem inappropriate, threatening, offensive,
or harmful.

Community leaders have the right and responsibility to remove, edit, or reject
comments, commits, code, wiki edits, issues, and other contributions that are
not aligned to this Code of Conduct, and will communicate reasons for moderation
decisions when appropriate.

## Scope

This Code of Conduct applies within all community spaces, and also applies when
an individual is officially representing the community in public spaces.
Examples of representing our community include using an official email address,
posting via an official social media account, or acting as an appointed
representative at an online or offline event.

## Enforcement

Instances of abusive, harassing, or otherwise unacceptable behavior may be
reported to the community leaders responsible for enforcement at
community@ente.io.
All complaints will be reviewed and investigated promptly and fairly.

All community leaders are obligated to respect the privacy and security of the
reporter of any incident.

## Attribution

This Code of Conduct is adapted from the [Contributor Covenant][homepage],
version 2.1, available at
[https://www.contributor-covenant.org/version/2/1/code_of_conduct.html][v2.1].

For answers to common questions about this code of conduct, see the FAQ at
[https://www.contributor-covenant.org/faq][FAQ]. Translations are available at
[https://www.contributor-covenant.org/translations][translations].

[homepage]: https://www.contributor-covenant.org
[v2.1]: https://www.contributor-covenant.org/version/2/1/code_of_conduct.html
[Mozilla CoC]: https://github.com/mozilla/diversity
[FAQ]: https://www.contributor-covenant.org/faq
[translations]: https://www.contributor-covenant.org/translations

**Comment by alexandre-walzberg** [comment]
Of course I tried "Favorite" and "Favorites" but no luck, it is not
recognized.

Le mar. 15 oct. 2024 à 06:21, Neeraj Gupta ***@***.***> a
écrit :
